sql/analyzer/rules.go

```diff
@@ -531,14 +534,82 @@ func resolveNaturalJoins(ctx *sql.Context, a *Analyzer, n sql.Node) (sql.Node, e
 
 		projections := append(append(common, left...), right...)
 
-		return plan.NewProject(
+		transformed = plan.NewProject(
 			projections,
 			plan.NewInnerJoin(
 				join.Left,
 				join.Right,
 				expression.JoinAnd(conditions...),
 			),
-		), nil
+		)
+
+		return transformed, nil
+	})
+
+	if err != nil || transformed == nil {
+		return node, err
+	}
+
+	var nodesToUnresolve = []sql.Node{}
+	plan.Inspect(node, func(iNode sql.Node) bool {
+		if iNode == transformed || iNode == nil {
+			return false
+		}
+
+		nodesToUnresolve = append(nodesToUnresolve, iNode)
+		return true
+	})
+
+	if len(nodesToUnresolve) == 0 {
+		return node, err
+	}
+
+	return node.TransformUp(func(node sql.Node) (sql.Node, error) {
+		var found bool
+		for _, nu := range nodesToUnresolve {
+			if nu.String() == node.String() {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			return node, nil
+		}
+
+		expressioner, ok := node.(sql.Expressioner)
+		if !ok {
+			return node, nil
+		}
+
+		return expressioner.TransformExpressions(func(e sql.Expression) (sql.Expression, error) {
+			var col, table string
+			switch e := e.(type) {
+			case *expression.GetField:
```

Comment:
lets merge these to cases into one:
`
case *expression.GetField, *expression.UnresolvedColumn:
`

Reply:
I can't. `e` would stay as a `sql.Expression` interface so I couldn't access `Name()` and `Table()` methods